js_src/rdm.js

```diff
@@ -1,146 +1,149 @@
-            var device3 = data[10] & data[11];
-
-            var device = (device0 << 24) + (device1 << 16) + (device2 << 8) + device3;
-
-            var recovered_checksum = ((data[12] & data[13]) << 8) +
-                (data[14] & data[15]);
-
-            var calculated_checksum = data.slice(0, 12).reduce(
-                function(previousValue, currentValue, index, array) {
-                  return previousValue + currentValue;
-                }
-            );
-
-            if (recovered_checksum != calculated_checksum) {
-                convertor.error = 'Checksum mismatch, was ' + recovered_checksum +
-                ', calculated to be ' + calculated_checksum;
-                return;
-            }
-
-            function zeroPad(num, places) {
-                var str = num.toString(16);
-                var zero = places - str.length + 1;
-                return Array(+(zero > 0 && zero)).join("0") + str;
-            }
-
-            convertor.uid = zeroPad(manufacturer) + ':' + zeroPad(device, 8);
-        };
-    }]);
+ .config(['$interpolateProvider', function ($interpolateProvider) {
+  'use strict';
+  $interpolateProvider.startSymbol('{[{');
+  $interpolateProvider.endSymbol('}]}');
+ }])
+ .controller('UIDController', ['$scope', function (convertor) {
+  'use strict';
+  convertor.euid = '';
+  convertor.error = '';
+  convertor.uid = '';
+
+  convertor.convertToEUID = function () {
+   convertor.euid = '';
+   convertor.error = '';
+
+   if (!convertor.uid.match(/^[0-9a-fA-F]{4}:[0-9a-fA-F]{8}$/)) {
+    convertor.error = 'Invalid UID';
+    return;
+   }
+   var tokens = convertor.uid.split(':');
+   if (tokens.length !== 2) {
+    convertor.error = 'Invalid UID';
+    return;
+   }
+
+   var manufacturer_str = tokens[0];
+   var device_str = tokens[1];
+   if (manufacturer_str.length !== 4 || device_str.length !== 8) {
+    convertor.error = 'Invalid UID';
+    return;
+   }
+
+   var manufacturer = parseInt(manufacturer_str, 16);
+   var device = parseInt(device_str, 16);
+   if (isNaN(manufacturer) || isNaN(device)) {
+    convertor.error = 'Invalid UID';
+    return;
+   }
+
+   var euid_bytes = [];
+
+   var manufacturer0 = (manufacturer >> 8) & 0xff;
+   var manufacturer1 = manufacturer & 0xff;
+   var device0 = (device >> 24) & 0xff;
+   var device1 = (device >> 16) & 0xff;
+   var device2 = (device >> 8) & 0xff;
+   var device3 = device & 0xff;
+
+   euid_bytes.push(manufacturer0 | 0xaa);
+   euid_bytes.push(manufacturer0 | 0x55);
+   euid_bytes.push(manufacturer1 | 0xaa);
+   euid_bytes.push(manufacturer1 | 0x55);
+
+   euid_bytes.push(device0 | 0xaa);
+   euid_bytes.push(device0 | 0x55);
+   euid_bytes.push(device1 | 0xaa);
+   euid_bytes.push(device1 | 0x55);
+   euid_bytes.push(device2 | 0xaa);
+   euid_bytes.push(device2 | 0x55);
+   euid_bytes.push(device3 | 0xaa);
+   euid_bytes.push(device3 | 0x55);
+
+   var checksum = euid_bytes.reduce(
+    function (previousValue, currentValue) {
```

Comment:
never mind you can disable it for function parameters but leave it active for var statements